### PR TITLE
[AutoMM] Add the few shot learner

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*

- Refactor previous `FewShotSVMPredictor` into `FewShotSVMLearner`. Now users can import `MultiModalPredictor` to do few shot learning. The API is unified and easier to use.
- Add new problem type `few_shot_classification` and the corresponding presets.
- Refactor the few shot tutorial accordingly, which is more clean and easier to read.
- Move some object detection specific attributes into the object detection learner.
- Remove the `few_shot_text_classification` problem type in favor of `few_shot_classification`.
- Fix multi-gpu training and prediction for the CLIP model.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/learners/base.py [hunk 1]
```diff
@@ -161,13 +161,10 @@ def __init__(
         hyperparameters: Optional[dict] = None,
```
Comment (suzhoum) [inline multimodal/src/autogluon/multimodal/learners/base.py:161]: Unrelated to this line, but do we want to add a default `eval_metric` for this problem type?

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/learners/base.py:161]: Good call! We can add the accuracy as the default.

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/learners/base.py:161]: Added accuracy as the default and tested in unit tests.

File: multimodal/src/autogluon/multimodal/learners/base.py [hunk 2]
```diff
@@ -193,7 +190,7 @@ def __init__(
             - 'image_text_similarity': Text-image similarity problem
             - 'feature_extraction': Extracting feature (only support inference)
             - 'zero_shot_image_classification': Zero-shot image classification (only support inference)
-            - 'few_shot_text_classification': (experimental) Few-shot text classification
+            - 'few_shot_classification': Few-shot classification for image or text data.
```
Comment (tonyhoo) [inline multimodal/src/autogluon/multimodal/learners/base.py:193]: [minor] does this work for matching and document classification?

**Review by tonyhoo — APPROVED** [review]

**Review by suzhoum — APPROVED** [review]
LGTM! Thanks for the change!

File: multimodal/src/autogluon/multimodal/learners/base.py (hunk 2 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/learners/base.py:194]: Not tested for document data yet.

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/learners/base.py:194]: Doesn't support matching.

**Pull request merged**